charts/argo-cd/values.yaml

```diff
@@ -2071,13 +2071,14 @@ server:
     ## Ref: https://argo-cd.readthedocs.io/en/stable/operator-manual/ingress/#aws-application-load-balancers-albs-and-classic-elb-http-mode
     aws:
       # -- Backend protocol version for the AWS ALB gRPC service
-      ## This tells AWS to send traffic from the ALB using HTTP2. Can use gRPC as well if you want to leverage gRPC specific features
-      backendProtocolVersion: HTTP2
+      ## This tells AWS to send traffic from the ALB using gRPC. Can use HTTP2 as well
+      ## For more information: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/target-group-health-checks.html#health-check-settings
+      backendProtocolVersion: GRPC
       # -- Service type for the AWS ALB gRPC service
       ## Can be of type NodePort or ClusterIP depending on which mode you are running.
       ## Instance mode needs type NodePort, IP mode needs type ClusterIP
-      ## Ref: https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.2/how-it-works/#ingress-traffic
-      serviceType: NodePort
+      ## Ref: https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.7/how-it-works/#ingress-traffic
+      serviceType: ClusterIP
```

Comment:
[ALB Guide](https://github.com/argoproj/argo-helm/tree/main/charts/argo-cd#aws-application-load-balancer) is also mention this, so I think it's reasonable to change default value, in addition to https://github.com/argoproj/argo-helm/issues/2552 . 🙋 
<img width="1139" alt="image" src="https://github.com/argoproj/argo-helm/assets/30188755/3261cbe6-23a3-47b1-907a-d4e60aac14b5">

Reply:
@yu-croco I would keep `NodePort` as is. The instance mode is default for AWS load balancer controller. The `ClusterIP` is needed only for `alb.ingress.kubernetes.io/target-type: ip`

Reply:
Thank you for your review.
Then what about adding comments for `IP mode` ? 🙋

Reply:
@yu-croco it's already there :)

```yaml
# -- Service type for the AWS ALB gRPC service
## Can be of type NodePort or ClusterIP depending on which mode you are running.
## Instance mode needs type NodePort, IP mode needs type ClusterIP
```

Reply:
Yes. `NodePort` means that LBs are forwarding traffic to all nodes and then it's routed to Pod via ingress controller. `ClusterIP` for IP mode means that LBs are forwarding traffic to AWS Elastic IP of Pod directly

Reply:
> it's already there :)

I see. 💡 

>The mode is just for serviceType- both of them can use backendProtocol HTTP2 or GRPC

In the original issue, https://github.com/argoproj/argo-helm/issues/2552, `HTTP2` causes problem for IP mode. 👀

Reply:
We can change this to `GRPC` as this service is technically gRPC running on port 443 and it's just protocol definition.

Reply:
The `serviceType` is usually important if you run with different CNI add-on such as Calico or Cilium. In overlay mode with masquerading you are restricted to instance mode.

Reply:
>>The mode is just for serviceType- both of them can use backendProtocol HTTP2 or GRPC
>
> In the original issue, https://github.com/argoproj/argo-helm/issues/2552, HTTP2 causes problem for IP mode. 👀

My bad, wrong quote. 😢 
It was about `backendProtocolVersion`

>Target health status of target group corresponding to this service becomes unhealthy because the value server.ingress.aws.backendProtocolVersion is HTTP2.
>It causes argocd-server pod's "not ready" status in environment which benefit from [pod readiness gate](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.7/deploy/pod_readiness_gate/).
>
>Target health status become healthy after setting server.ingress.aws.backendProtocolVersion to GRPC.

Reply:
Hi @mikutas , if you have comments, please follow up.
*I'm sorry for the mess...

Reply:
@pdrastil As [reported](https://github.com/argoproj/argo-helm/issues/2552), if protocol version of target group (for gRPC Service) is HTTP2, the health check setting of the TG become invalid and target become unhealthy.
So aws.backendProtocolVersion is not selectable indeed.
I think this is the point of this issue.

On the other hand, `serviceType` is certainly selectable corresponding to AWS LBC's mode(instance / ip), and it's irrelevant to the issue of unhealthy target.
Therefore, I think the default value of `serviceType` need not to be changed in this PR and keeping `NodePort` as default is reasonable as well as AWS LBC's default mode is `instance`.

Reply:
From my experience with ALB (and AWS EKS) serviceType is for sure relevant.
If you use `ClusterIP` and ALB target-type `ip` the health checks are handled different than in `NodePort` together with `instance`.

- `instance` by default checks traffic path until nodes
- `ip` always tries to check directly to the pod and uses `/` and "traffic port" as the default health-checking parameters

Reply:
@mikutas Thanks for response. @yu-croco Changing the port to GRPC without any changes to service type should be enough.

Reply:
@mkilchhofer Thank you, my understanding seems incorrect.
More precicely speaking:
- `backendProtocolVersion` = `HTTP2` and `serviceType` = `ClusterIP`  (i.e. LBC is `ip` mode) -> unhealthy (my case)
- `backendProtocolVersion` = `HTTP2` and `serviceType` = `NodePort` (i.e. LBC is `instance` mode) -> healthy (your case)
- `backendProtocolVersion` = `GRPC` and `serviceType` = `ClusterIP` (i.e. LBC is `ip` mode) -> healthy (my case)
- `backendProtocolVersion` = `GRPC` and `serviceType` = `NodePort` (i.e. LBC is `instance` mode) -> healthy (probably?)

Reply:
>Changing the port to GRPC without any changes to service type should be enough.

@pdrastil @mikutas 
I'm sorry for my mess... I reverted unnecessary diff in https://github.com/argoproj/argo-helm/pull/2553/commits/f91e531fdc6c0322a189ae51632a21979024b611 . 🙋